extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java

```diff
@@ -659,8 +660,16 @@ protected void updateChannel(String channeldD) {
                 url = getAlbumArtUrl();
                 if (url != null) {
                     try {
-                        InputStream input = url.openStream();
-                        newState = new RawType(IOUtils.toByteArray(input));
+                        URLConnection connection = url.openConnection();
+                        String contentType = connection.getContentType();
+                        InputStream input = connection.getInputStream();
+                        if (contentType == null) {
+                            contentType = URLConnection.guessContentTypeFromStream(input);
```

Comment:
This method requires "an input stream that supports marks" - otherwise, an exception will be thrown (because the stream has to be reset).
You will have to catch this exception and potentially (?) close and re-create the input stream.
Did you test whether this method works at all for media content?